prov/gni/src/gnix_cm_nic.c

```diff
@@ -236,13 +237,17 @@ int _gnix_cm_nic_progress(struct gnix_cm_nic *cm_nic)
 	 */
 
 	if (cm_nic->ctrl_progress == FI_PROGRESS_MANUAL) {
-		ret = _gnix_dgram_poll(cm_nic->dgram_hndl,
-					  GNIX_DGRAM_NOBLOCK);
-		if (ret != FI_SUCCESS) {
-			GNIX_WARN(FI_LOG_EP_CTRL,
-				"_gnix_dgram_poll returned %s\n",
-				  fi_strerror(-ret));
-				goto err;
+		++cm_nic->poll_cnt;
+		if (((cm_nic->poll_cnt % 512) == 0)  ||
+			!dlist_empty(&cm_nic->cm_nic_wq)) {
```

Comment:
If the cm_nic->cm_nic_wq is empty more often than the cm_nic->poll_cnt reaches 512, would it be faster to short circuit this if statement by moving !dlist_empty(&cm_nic->cm_nic_wq) to the left of the logical or?